osmnx/io.py

```diff
@@ -421,6 +421,16 @@ def _convert_node_attr_types(G, dtypes=None):
     G : networkx.MultiDiGraph
     """
     for _, data in G.nodes(data=True):
+        # first, eval stringified lists, dicts or sets to convert them to objects
+        # edge attributes might have a single value, or a list if simplified
```

Comment:
This comment is specific to edge attributes, but this code block concerns node attributres.

Reply:
Whoops!

Reply:
Fixed.  Should I try to address the code coverage issue with additional tests?

Reply:
Yes please add a test for this new functionality.

Reply:
I've modified test_osmnx.py to add tests for list, set, dict attributes for both nodes and edges, if that scope change is ok. 

A local test, running `pytest --cov=./osmnx --cov-report=term-missing --verbose` in a terminal on the local git code shows all tests are passing, so the testing code changes appear to be working:

      tests/test_osmnx.py::test_logging PASSED                                 [  7%]
      tests/test_osmnx.py::test_coords_rounding PASSED                         [ 15%]
      tests/test_osmnx.py::test_geocoder PASSED                                [ 23%]
      tests/test_osmnx.py::test_stats PASSED                                   [ 30%]
      tests/test_osmnx.py::test_osm_xml PASSED                                 [ 38%]
      tests/test_osmnx.py::test_elevation PASSED                               [ 46%]
      tests/test_osmnx.py::test_routing PASSED                                 [ 53%]
      tests/test_osmnx.py::test_plots PASSED                                   [ 61%]
      tests/test_osmnx.py::test_find_nearest PASSED                            [ 69%]
      tests/test_osmnx.py::test_endpoints PASSED                               [ 76%]
      tests/test_osmnx.py::test_graph_save_load PASSED                         [ 84%]
      tests/test_osmnx.py::test_graph_from_functions PASSED                    [ 92%]
      tests/test_osmnx.py::test_features PASSED                                [100%]

However, there was a warning immediately afterward the above: `CoverageWarning: No data was collected.`, which I'm not sure how to interpret.

Separately, format.sh and lint_test.sh were run but no apparent automatic changes to the file, so I will push it to my fork, and I suppose the test_osmnx.py file will get added to this PR automatically in addition to the io.py file ...